cloud/aws/cli/core.py

```diff
@@ -141,16 +148,37 @@ def current_lambda_image(c, repo_arn):
     return current
 
 
-@task()
+@task
 def deploy_lambda_image(c):
     """Build and push the lambda image, fails if step 1 is not deployed"""
-    image_url = terraform_output(c, "step-1", "vast_lambda_repository_url")
+    image_url = terraform_output(c, "core-1", "vast_lambda_repository_url")
+    repo_arn = terraform_output(c, "core-1", "vast_lambda_repository_arn")
+    # get the digest of the current image
+    current_image = current_lambda_image(c, repo_arn)
+    try:
+        c.run(f"docker pull {current_image}")
+        old_digest = c.run(
+            f"docker inspect --format='{{{{.RepoDigests}}}}' {current_image}",
+            hide="out",
+        ).stdout
+    except:
+        old_digest = "current-image-not-found"
```

Comment:
@dispanser do you find that it is hurting readability?

Reply:
Well, not necessarily readability but re-using some variable to hold very different things hurts the Rust programmer inside of me (cough, `Result<Url, Error>`, cough). I can live with this because it's contained in that single function, and is only used as a baseline for `new_digest`, so no action needed.